benchmarks/src/test/java/io/crate/data/join/RowsBatchIteratorBenchmark.java

```diff
@@ -114,4 +114,21 @@ public void measureConsumeNestedLoopLeftJoin(Blackhole blackhole) throws Excepti
         }
         leftJoin.moveToStart();
     }
+
+    @Benchmark
+    public void measureConsumeHashInnerJoin(Blackhole blackhole) throws Exception {
+        BatchIterator<Row> leftJoin = JoinBatchIterators.hashInnerJoin(
+            InMemoryBatchIterator.of(oneThousandRows, SENTINEL),
+            InMemoryBatchIterator.of(tenThousandRows, SENTINEL),
+            new CombinedRow(1, 1),
+            row -> Objects.equals(row.get(0), row.get(1)),
+            row -> Objects.hash(row.get(0)),
```

Comment:
I would add one more test to simulate hash collisions by changing the hash builder functions similar to: https://github.com/crate/crate/commit/aad97ce0345b8481c3f7d567540367d56ec90724#diff-587d834e25d4158dd704f5d76a033363R64